src/TrackableTrait.php

```diff
@@ -40,7 +40,19 @@ trait TrackableTrait
      */
     public function getDesiredName(): string
     {
-        return preg_replace('/.*\\\\/', '', mb_strtolower(static::class));
+        // can be anything, but better to build meaningful name
+        $name = static::class;
+        if (strpos($name, 'class@anonymous') === 0) {
+            $name = '';
+            foreach (class_parents(static::class) as $name) {
```

Comment:
I do not exactly figure this foreach loop. To my understanding, `$name` defined in `as $name` should be only available inside the foreach loop and have nothing to do with the outer `$name` var. Or am I wrong here and the outer `$name` is changed within this foreach? Because If not I dont see what the foreach is doing

Reply:
I improved it now, it worked but was not that clean. (and it was actually wrong for nested anonymous classes, but they are not supported by PHP as they can not be created by the current PHP grammar)